src/services/completions.ts

```diff
@@ -675,6 +675,37 @@ namespace ts.Completions {
             hasAction = !importCompletionNode;
         }
 
+        const kind = SymbolDisplay.getSymbolKind(typeChecker, symbol, location); // TODO: GH#18217
+        if (kind === ScriptElementKind.jsxAttribute && preferences.jsxSnippetCompletion && preferences.jsxSnippetCompletion !== "none") {
+            let useBraces = preferences.jsxSnippetCompletion === "braces";
+            const type = typeChecker.getTypeOfSymbolAtLocation(symbol, location);
+
+            // If is boolean like or undefined, don't return a snippet we want just to return the completion.
+            if (preferences.jsxSnippetCompletion === "auto"
+                && !(type.flags & TypeFlags.BooleanLike)
+                && !(type.flags & TypeFlags.Union && every((type as UnionType).types, type => !!(type.flags & (TypeFlags.BooleanLike | TypeFlags.Undefined))))

```

Comment:
You might want to check if _any_ type is a boolean, since that’s all it takes to make `<Foo prop />` valid.

Reply:
I think I prefer that to show the shorthand it must be only `boolean` and braces for everything else. This is true for a type `boolean | string` for example, which will use braces as well.

Nonetheless, I think there's a argument for adding as little as possible, as in the case of if _any_ of the types is boolean. If you feel this is a better approach I'm don't mind changing it.

Reply:
I do think that’s better, since a user’s baseline expectation is just to complete the prop name. Users can’t really complain (famous last words) if we don’t complete the `={}` because we _already_ don’t complete that. It’s like an added bonus. But users will definitely complain if they frequently find themselves having to delete characters we inserted for them.

Reply:
Makes sense. Changed it.